src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/HttpApi/ApiClient.cs

```diff
@@ -408,5 +392,45 @@ private static void AppendDuration(StringBuilder builder, TimeSpan duration)
                 builder.Append(Convert.ToInt32(duration.TotalSeconds).ToString(CultureInfo.InvariantCulture));
             }
         }
+
+        private static string GetProcessQuery(int? pid = null, Guid? uid = null, string name = null)
+        {
+            string assembledQuery = "";
+            bool initialQuery = true;
+
+            if (pid != null)
+            {
+                assembledQuery += "pid=" + pid.GetValueOrDefault().ToString(CultureInfo.InvariantCulture);
+                initialQuery = false;
+            }
+
+            if (uid != null)
+            {
+                if (initialQuery == false)
+                {
+                    assembledQuery += "&";
+                }
+
+                assembledQuery += "uid=" + uid.GetValueOrDefault().ToString("D");
+                initialQuery = false;
+            }
+
+            if (name != null)
+            {
+                if (initialQuery == false)
+                {
+                    assembledQuery += "&";
+                }
+
+                assembledQuery += "name=" + name;
+            }
+
+            if (!assembledQuery.Equals(""))
+            {
+                return assembledQuery;
+            }
+
+            throw new Exception("One of PID, UID, or Name must be specified.");
```

Comment:
Try to use more exceptions rather than Exception itself e.g. ArgumentException would be fitting here.